flight/PiOS/Common/pios_dsm.c

```diff
@@ -316,9 +316,40 @@ int32_t PIOS_DSM_Init(uintptr_t *dsm_id,
 	/* Bind the configuration to the device instance */
 	dsm_dev->cfg = cfg;
 
+	uint8_t num_pulses = 0;
+
+	// check user settings to determine which resolution mode to use
+	// if invalid mode selected, bail
+	if (mode > HWSHARED_DSMXMODE_BIND10PULSES)
```

Comment:
Basically all variants of these patterns are bad in one way or another.

1. The case where we have a default prevents the compiler from emitting a message about uncovered case elements.  That warning (that becomes an error) is super nice which is why I'm not a fan of default in general.
2. Enums aren't guaranteed in C to be in range.  They're more like "a collection of more tightly scoped constants."  There's been a long history of using them for flags, which the standard recognizes.  e.g. 
        enum flags { A=1, B=2, C=4, D=8 };

        enum flags test = A | B;

        printf("%d\n", test);
is completely legal code.  All the compiler promises is to pick a underlying type that is big enough to store all the enumeration values, and to give us warnings if we don't cover given values in switch statements.  (It's assumed if we're using the enum in a switch we really mean a true enumeration of possible values).
3. If we hit an uncovered case-- an invalid enum value with missing default, undefined behavior occurs.  So really we need to validate enum ranges somewhere.
4. Do we do this loading from flash / over the telemetry link?  That would be best rather than contaminating our code with lots of checks all over the place.

So my end thought is "meh."  There's flaws with just about anything you can do.  Also, I strongly believe in #1792 .  If we do validation, we should put it in a sufficient number of places to catch all the inputs to the code, and then we should A) not spend a lot of effort checking values afterwards, and B) where we do sprinkle in assertions afterwards, they should be fatal / cause the system to fail fast rather than do undefined things.